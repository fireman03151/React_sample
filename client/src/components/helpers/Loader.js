import React from 'react';
import PropTypes from 'prop-types';
import Spinner from 'react-spinkit';

import './loader.css';

function Loader({ fullScreen }) {
  return (
    <div className={fullScreen ? 'full-screen-wrapper' : ''}>
      <Spinner name='line-scale-pulse-out' />
    </div>
  );
}

Loader.displayName = 'Loader';
Loader.propTypes = {
  fullScreen: PropTypes.bool
};

export default Loader;
