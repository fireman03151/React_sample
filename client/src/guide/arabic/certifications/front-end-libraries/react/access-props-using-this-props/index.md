---
title: Access Props Using this.props
localeTitle: وصول الدعائم باستخدام هذا. المحاصيل
---
## \## Access Props باستخدام هذا .props

للحصول على هذا الحل ، يجب أن تتذكر أولاً كيفية إضافة الأدوات الدعائية إلى المكون الرئيسي:

\`\` \`JSX

 `Once you have the prop set, you can use this.props in your child component. 
` 

أتش تي أم أل

كلمة المرور المؤقتة هي: **{this.props.tempPassword}**

\`\` \`