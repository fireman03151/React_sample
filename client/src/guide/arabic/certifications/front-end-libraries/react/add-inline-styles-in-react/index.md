---
title: Add Inline Styles in React
localeTitle: إضافة أنماط مضمنة في React
---
## إضافة أنماط مضمنة في React

يمكنك تعريف نمط مكون تمرير الكائن مباشرة كـ "نمط" prop. فقط تذكر أن كل خاصية من كائن النمط هي camelcased. لذا ، تُعلن الخصائص مثل "font-size" بأنها "fontSize" لتكون خاصية كائن جافا سكريبت صالحة.

### المفسد

 `class Colorful extends React.Component { 
  render() { 
    // change code below this line 
    return ( 
      <div style={{color: "yellow", fontSize: 24}}>Style Me!</div> 
    ); 
    // change code above this line 
  } 
 }; 
` 

### مصادر

[DOM Elements Style](https://reactjs.org/docs/dom-elements.html#style)