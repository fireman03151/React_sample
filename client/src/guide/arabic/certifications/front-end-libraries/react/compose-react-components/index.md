---
title: Compose React Components
localeTitle: تأليف React Components
---
## تأليف React Components

### ملحوظة

استخدام المكونات المتداخلة كما في challemge السابقة لتقديم المكونات.

### حل

ما يلي هو الحل لل chakkenge ، حيث تجعل الحمضيات و NonCitrus في مكون ثم يتم تقديمها في آخر:

 `class Fruits extends React.Component { 
  constructor(props) { 
    super(props); 
  } 
  render() { 
    return ( 
      <div> 
        <h2>Fruits:</h2> 
        <NonCitrus /> 
        <Citrus /> 
      </div> 
    ); 
  } 
 }; 
 
 class TypesOfFood extends React.Component { 
  constructor(props) { 
     super(props); 
  } 
  render() { 
    return ( 
      <div> 
        <Fruits /> 
        <Vegetables /> 
      </div> 
    ); 
  } 
 }; 
` 

### روابط ذات صلة:

*   [المكونات والدعائم](https://reactjs.org/docs/components-and-props.html)
*   [المكونات المتداخلة](http://www.reactjstutorial.net/nested-components.html)