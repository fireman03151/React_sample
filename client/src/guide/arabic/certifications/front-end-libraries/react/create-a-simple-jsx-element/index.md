---
title: Create a Simple JSX Element
localeTitle: إنشاء عنصر بسيط JSX
---
## إنشاء عنصر بسيط JSX

هذا التحدي بسيط إلى حد ما ، واستبدل div في عنصر JSX مع h1 `javascript const JSX=<h1></h1>;` ثم أضف النص المطلوب. `javascript const JSX = <h1>Hello JSX!</h1>;`