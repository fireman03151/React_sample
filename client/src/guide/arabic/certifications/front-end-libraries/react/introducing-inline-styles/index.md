---
title: Introducing Inline Styles
localeTitle: إدخال الأنماط الداخلية
---
## إدخال الأنماط الداخلية

## حل

يمكن أن يكون هذا الأمر صعبًا قليلاً لأن JSX يشبه إلى حد بعيد HTML ولكن **ليس نفسه** .

لنستعرض الخطوات حتى تعرف الفرق. قم أولاً بتعيين علامة النمط على **كائن JavaScript** .

 `class Colorful extends React.Component { 
  render() { 
    return ( 
      <div style={{}}> 
        Big Red 
      </div> 
    ); 
  } 
 }; 
` 

الآن لديك تعيين علامة النمط الخاص بك إلى كائن فارغ. لاحظ كيف أن هناك مجموعتين من الأقواس المتعرجة. هذا اختلاف مهم بين JSX و HTML.

ثانيًا ، لنقم بتعيين اللون إلى اللون الأحمر.

 `class Colorful extends React.Component { 
  render() { 
    return ( 
      <div style={{ color: 'red' }}> 
        Big Red 
      </div> 
    ); 
  } 
 }; 
` 

أخيرًا ، لنقم بتعيين حجم الخط إلى 72 بكسل.

### المفسد

 `class Colorful extends React.Component { 
  render() { 
    return ( 
      <div style={{ color: 'red', fontSize: '72'}}> 
        Big Red 
      </div> 
    ); 
  } 
 }; 
` 

لاحظ كيف أن سمة **JSX** هي **camelCase** . هذا اختلاف مهم آخر يجب تذكره حول JSX. بالإضافة إلى ذلك ، ربما لاحظت عدم وجود وحدة. في JSX ، عند تعيين سمة fontSize ، تكون **الوحدة اختيارية** وسيتم تعيينها تلقائيًا على px إذا لم يتم تعيينها يدويًا.