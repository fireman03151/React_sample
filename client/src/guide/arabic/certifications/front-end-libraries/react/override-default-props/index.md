---
title: Override Default Props
localeTitle: تجاوز الدعائم الافتراضية
---
## تجاوز الدعائم الافتراضية

لقد تجاوزت هذا التحدي القيمة الافتراضية `quantity` الدعائية لمكون العناصر. حيث يتم تعيين القيمة الافتراضية `quantity` إلى `0` .

 `const Items = (props) => { 
  return <h1>Current Quantity of Items in Cart: {props.quantity}</h1> 
 } 
 
 Items.defaultProps = { 
  quantity: 0 
 } 
` 

لتجاوز قيمة الدعائم الافتراضية ، تكون الصيغة التي يجب اتباعها هي

 `<Component propsName={Value}/> 
` 

باتباع "بناء الجملة" ، يجب أن يتم تعريف التعليمة البرمجية التالية أدناه التعليمات البرمجية المحددة

 `<Items quantity={50}/> 
` 

سيؤدي هذا إلى تجاوز القيمة من `0` إلى `50`