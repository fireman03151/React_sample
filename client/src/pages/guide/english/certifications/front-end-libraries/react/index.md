---
title: React
---
## React

![React](https://upload.wikimedia.org/wikipedia/commons/thumb/a/a7/React-icon.svg/200px-React-icon.svg.png)

React or ReactJS is a front-end library which is used to create beautiful, reusable and interactive User Interfaces.

React is written in JavaScript and JSX. It is designed with reusability in mind and follows a component based pattern, which means write once and use infinitely.

It is available as a CDN for browser based development or as a npm package for Command line shell based editing.

React has been developed under the Facebook Open Source project, and is actively maintained by the Open Source community.

#### More Information:
<!-- Please add any articles you think might be helpful to read before writing the article -->
1. [React](https://reactjs.org/)
2. [React Open Source page](https://github.com/facebook/react/)
3. [React Forum](https://discuss.reactjs.org/)
