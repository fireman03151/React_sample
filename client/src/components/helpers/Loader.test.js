/* global expect */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';

import Loader from './Loader';

describe('<Loader />', () => {
  afterEach(cleanup);

  it('renders to the DOM', () => {
    const { container } = render(<Loader />);
    expect(container).toBeTruthy();
  });

  it('adds the correct class when given a fullScreen prop', () => {
    const { container } = render(<Loader fullScreen={true} />);
    expect(container.firstChild).toHaveClass('full-screen-wrapper');
  });

  /**
   * As we only wrap another library Component,
   * there is nothing much to test except snapshots
   */

  it('matches to the default render snapshot', () => {
    const { container } = render(<Loader />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches the fullScreen render snapshot', () => {
    const { container } = render(<Loader fullScreen={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
