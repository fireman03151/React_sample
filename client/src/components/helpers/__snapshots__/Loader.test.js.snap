// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> matches the fullScreen render snapshot 1`] = `
<div
  class="full-screen-wrapper"
>
  <div
    class="sk-fade-in sk-spinner line-scale-pulse-out"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;

exports[`<Loader /> matches to the default render snapshot 1`] = `
<div
  class=""
>
  <div
    class="sk-fade-in sk-spinner line-scale-pulse-out"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
